result_collector.go

```diff
@@ -0,0 +1,137 @@
+package wire
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+)
+
+// Pipeline safety limits
+const (
+	MaxPipelineQueries = 20
+	MaxPipelineRows    = 50000
+)
+
+// ResultCollector implements DataWriter interface
+// It collects query results in memory for later replay
+type ResultCollector struct {
+	mu      sync.Mutex
```

Comment:
Is a mutex needed since the connection is consumed sequentially?

Reply:
It is not, this snuck in from an earlier version of this implementation - removed 👍